osu.Game/Screens/Play/SpectatorPlayer.cs

```diff
@@ -79,11 +80,14 @@ private void userSentFrames(int userId, FrameDataBundle bundle)
             if (!this.IsCurrentScreen())
                 return;
 
+            if (GameplayState.Ruleset is not ILegacyRuleset legacyRuleset)
+                return;
```

Comment:
Probably should assert instead, since it currently shouldn't be possible for a player with a custom ruleset to broadcast frames to the server as far as I'm aware?